src/app/api/v1/endpoints/extract_data.py

```diff
@@ -189,11 +192,14 @@ def extract_text(pdf_page: fitz.Page, user_defined_bbox: fitz.Rect) -> ExtractDa
         text += text_line.text + " "
         text_based_bbox = text_based_bbox + text_line.rect
 
-    bbox = BoundingBox.load_from_fitz_rect(text_based_bbox)
-    return ExtractTextResponse(bbox=bbox, text=text)
+    if text:
+        bbox = BoundingBox.load_from_fitz_rect(text_based_bbox)
+        return ExtractTextResponse(bbox=bbox, text=text)
+    else:
+        raise HTTPException(status_code=404, detail="Text not found.")
 
 
-def extract_number(pdf_page: fitz.Page, user_defined_bbox: fitz.Rect) -> ExtractNumberResponse:
+def extract_number(pdf_page: fitz.Page, user_defined_bbox: fitz.Rect) -> ExtractNumberResponse | HTTPException:
```

Comment:
Can be reverted to just `ExtractNumberResponse` as return type?

Reply:
Exactly!